code/api/src/main/scala/com/avast/datadog4s/api/tag/TagValue.scala

```diff
@@ -0,0 +1,15 @@
+package com.avast.datadog4s.api.tag
+
+trait TagValue[A] { self =>
+  def convert(a: A): String
+
+  def contramap[B](f: B => A): TagValue[B] = (a: B) => self.convert(f(a))
```

Comment:
We should consider to implement `Contravariant` typeclass from cats instead. The only issue is that it adds cats dependency to api. Or we could provide integration via separate separate module.

Reply:
i don't think it's worth it at this point